src/github_runner_manager/metrics/storage.py

```diff
@@ -48,145 +48,222 @@ class StorageManager(Protocol):  # pylint: disable=too-few-public-methods
         list_all: Method to list all storages.
         get: Method to get a storage by name.
         delete: Method to delete a storage by name.
+        move_to_quarantine: Method to archive and delete a storage by name.
     """
 
     create: Callable[[str], MetricsStorage]
     list_all: Callable[[], Iterator[MetricsStorage]]
     get: Callable[[str], MetricsStorage]
     delete: Callable[[str], None]
+    move_to_quarantine: Callable[[str], None]
 
 
-def _get_runner_fs_path(runner_name: str) -> Path:
-    """Get the path of the runner shared filesystem.
+class StorageManager(StorageManagerProtocol):
+    """Manager for the metrics storage."""
 
-    Args:
-        runner_name: The name of the runner.
+    def __init__(self, system_user_config: SystemUserConfig):
+        """Initialize the storage manager.
 
```

Comment:
We move the implementation to a class, as all functions will require the system_user to set up/get the metrics store accordingly. 
We would also need to pass the system config user to the extract method of `metrics.runner`, and I think having it in the class is a better abstraction.